src/deepsparse/v2/text_generation/multi_engine_prefill_operator.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from enum import Enum
+from typing import Any, Optional
+
+import numpy
+
+from deepsparse.transformers.utils.helpers import create_causal_mask
+from deepsparse.v2.operators import Operator
+from deepsparse.v2.utils import Context, InferenceState, PipelineState
+
+
+__all__ = ["MultiEnginePrefill"]
+
+
+class OnnxInputNames(Enum):
+    INPUT_IDS = "input_ids"
+    ATTN_MASK = "attention_mask"
+    CAUSAL_MASK = "causal_mask"
+    POSITIONS = "positions"
+
+
+# NOTE: A possible clean-up could involve combining this Operator and the
```

Comment:
yes, let's do this if possible as discussed, especially if we can share logic and minimize the branching points in the graph

Reply:
This is what #1364 does :)

Reply:
Oh but I see, you want to go one step further and combine both operators. Cool.

Reply:
I think lets start by adding in Damian's change to clean-up. We can go a step further to combine the operators after.